dexlib2/src/main/java/org/jf/dexlib2/builder/LocatedItems.java

```diff
@@ -56,32 +56,37 @@
                 };
             }
 
-            @Override public int size() {
-                return getImmutableItems().size();
+            @Override
+            public int size() {
+                return getItems().size();
             }
 
-            @Override public boolean add(@Nonnull T item) {
+            @Override
+            public boolean add(@Nonnull T item) {
                 if (item.isPlaced()) {
-
-
-                    throw new IllegalArgumentException(addLocatedItemError());
+                    throw new IllegalArgumentException(getAddLocatedItemError());
                 }
                 item.setLocation(newItemsLocation);
-                getMutableItems().add(item);
+                initItemsIfNull();
```

Comment:
Hmm. I'm still not a huge fan of this.

What do you think about adding a private addItem method, and doing the null check and initialization there?

As far as I can tell, the only case that doesn't directly cover is the call to [debugItems.addAll()](https://github.com/JesusFreke/smali/blob/90db3a16b79470f4b67a4e3c008f98f9c90e0326/dexlib2/src/main/java/org/jf/dexlib2/builder/MethodLocation.java#L118), but we can either add an addAllItems() method, or just iterate over the other's debug items call addItem() on each individually.

Reply:
I think your suggestion improves because it keeps the abstraction level of the anonymous class uniform.